artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTSubscriptionManager.java

```diff
@@ -301,9 +304,9 @@ private SimpleString getQueueNameForTopic(String topic) {
          int slashIndex = topic.indexOf(SLASH) + 1;
          String sharedSubscriptionName = topic.substring(slashIndex, topic.indexOf(SLASH, slashIndex));
          String parsedTopicName = topic.substring(topic.indexOf(SLASH, slashIndex) + 1);
-         return new SimpleString(sharedSubscriptionName).concat(".").concat(parsedTopicName);
+         return new SimpleString(sharedSubscriptionName).concat(delimiter).concat(parsedTopicName);
       } else {
-         return new SimpleString(session.getState().getClientId()).concat(".").concat(topic);
+         return new SimpleString(session.getState().getClientId()).concat(delimiter).concat(topic);
```

Comment:
if you could add some testing? a simple one is fine?

Also.. I would update versions.md, in case someone is updating the delimiter, with the fact that MQTT subscription now are using the configured delimiter as it was missed before because of "ARTEMIS-4350".

if you don't want to update the versions.md that's fine and I would do it myself.. but if you could at least add a test?